app/models/entities/TooComplexQueryError.scala

```diff
@@ -7,7 +7,11 @@ import play.api.Logging
 import sangria.execution.{ExceptionHandler, HandledException, MaxQueryDepthReachedError}
 import sangria.marshalling.ResultMarshaller
 
-case object TooComplexQueryError extends Exception("Query is too expensive.") with Logging {
+case object TooComplexQueryError
+    extends Exception(
+      "Query is too expensive. The response size is likely to be too large. Try requesting smaller page sizes"
```

Comment:
I'm not too sure about this bit Try requesting smaller page sizes because it would suggest that the queries have pagination and reducing the size could solve the issue and that's not always the case

Reply:
I think that looks good